cpp/src/io/parquet/reader_impl_helpers.cpp

```diff
@@ -536,6 +555,305 @@ aggregate_reader_metadata::aggregate_reader_metadata(
       CUDF_EXPECTS(schema == pfm.schema, "All sources must have the same schema");
     }
   }
+
+  if (arrow_schema.has_value()) {
+    // consume arrow schema into the parquet schema
+    consume_arrow_schema();
+    // we no longer keep arrow schema alive
+    arrow_schema.reset();
+  }
+
+  // erase "ARROW:schema" from the output pfm if exists
+  std::for_each(
+    keyval_maps.begin(), keyval_maps.end(), [](auto& pfm) { pfm.erase("ARROW:schema"); });
```

Comment:
why is this done?

Reply:
To ensure that this optional key value metadata doesn't make its way to the cudf table and isn't written to output parquet as the table might have changed by then. A new arrow:schema should be optional generated by parquet writer and written to file.